### ISSUE TITLE
Windows.Detection.NamedPipes doesn't work

### ISSUE THREAD

**Issue by Chaoui-lpb** [mgreen27/DetectRaptor#78]
Hello,

I don't often open issues on github, so please forgive me in advance if I'm unclear. In any case, here's my configuration, I have a Windows 10 endpoint on Virtualbox with a server running on my Ubuntu host. My aim is to detect whether the windows guest is infected by a cobalt strike beacon or not. To do this, I generate SMB cobalt strike beacons, which means that I have a pipe called msagent_33 on my winwdows machine. I check that it exists by doing a `Get-ChildItem \\.\pipe\` in powershell.

This is where my problem arises, when I run the VQL query Windows.Detection.NamedPipes the output gives me no result. What's more, there are no real errors in the logs apart from ‘GetObjectName status STATUS_NOT_SUPPORTED’ but I'm pretty sure that's not where it's coming from.

![Pasted image 20241028113245](https://github.com/user-attachments/assets/fc6913df-0102-4c18-8dd5-ad2f13839a15)

The reason I'm pretty sure the origin isn't in this error is when I want to run a Windows.System.Handles module on the name of my smb beacon. As shown in the screenshot below, I'm supposed to get the handle name but it doesn't show up. However, I'm sure that the name of this handle corresponds to the name of the pipe I'm looking for.

![Pasted image 20241028110655](https://github.com/user-attachments/assets/9f32e931-01d6-44a1-ad69-b64914882f6e)

In short, I think that the VQL script or my version of Velociraptor is unable to retrieve or read the names of the pipes present on the Windows machine.
Am I the only one with this problem? Has anyone managed to get this request to work?

Thanks in advance.

**Comment by mgreen27** [comment]
The named pipes artifacts has specific pipes it is looking for: https://github.com/mgreen27/DetectRaptor/blob/master/vql/NamedPipes.yaml#L18-L73

Can you rerun the artifact and use a . in the IoxRegex. Confirm if your getting results.

Also you can install and check sysmon for event id 17|18 (You might need to rerun your binary to check if your expected pipe is being created)?

I would expect the name of your pipe to show as: \Device\NamedPipe\NAMEOFPIPE
You can also confirm that you can view lsass or another well known pipe to validate.

**Comment by Chaoui-lpb** [comment]
Hello,

Thank you for your quick reply. As I'm just starting out, I think I've got a lot of basic things mixed up. I had installed sysmon incorrectly. By reinstalling sysmon with [a correct configuration file](https://github.com/olafhartong/sysmon-modular/blob/master/sysmonconfig.xml), the artefact was able to take effect.

Thanks a lot!
